src/utils/include/utils/LCA.hpp

```diff
@@ -167,18 +169,21 @@ inline int LCA::getLCA( int u, int v ) const
     return m_euler[queryRMQ( m_firstAppearance[u], m_firstAppearance[v] )];
 }
 
-// TODO: this function should be const! For that, we cannot use operator[] for the maps. They
-// are not const and can potentially modify the maps. We should maybe call m_encode.at(...).
-inline std::string LCA::getLCA( const std::vector< std::string >& taxIds )
+inline std::string LCA::getLCA( const std::vector< std::string >& taxIds ) const
 {
-    assert( taxIds.size() > 1 ); // I wonder if, aside from asserting this, if we should just branch off...
```

Comment:
I would this leave this assert, as we really expect `taxIds` to be at least `size == 2`. The assert will be there only on Debug mode, so it will not interfere on Release (the compiler will remove that automatically on Release)

Reply:
Here the best would be to return itself if size()==1 but I already check that before, not the best practice but I avoid all I can here

Reply:
Let's leave the assertion then... it will easily flag the intent and bears no performance cost at all